nodes/widgets/ui_button_group.js

```diff
@@ -9,9 +9,31 @@ module.exports = function (RED) {
         // which group are we rendering this widget
         const group = RED.nodes.getNode(config.group)
 
+        // Keep the code of this function in sync with the client-side function
+        function findOptionByValue (val) {
+            const opt = config.options?.find(option => {
+                if (typeof (val) === 'object') {
+                    return (JSON.stringify(val) === JSON.stringify(option.value))
+                } else {
+                    return option.value === val
+                }
+            })
+            if (opt) {
+                return opt
+            }
+            return null
+        }
+
         const evts = {
             onChange: true,
             beforeSend: function (msg) {
+                if (typeof msg.payload !== 'undefined') {
+                    const option = findOptionByValue(msg.payload)
+                    if (option) {
+                        node.status({ fill: 'blue', shape: 'dot', text: option.label })
```

Comment:
```suggestion
                        node.status({ fill: 'blue', shape: 'dot', text: option.label || option.value })
```

In cases where we have an icon-only button group, there is no `option.label`, so best to resort back to `option.value` rather than show nothing